stdlib/3.3/ipaddress.pyi

```diff
@@ -1,201 +1,150 @@
-from typing import Any, Iterable, Iterator, Optional, SupportsInt, Tuple, TypeVar, Union
+import sys
+from typing import (Any, Container, Generic, Iterable, Iterator, Optional,
+                    overload, SupportsInt, Tuple, TypeVar, Union)
 
-IPV4LENGTH = ...  # type: int
-IPV6LENGTH = ...  # type: int
+# Undocumented length constants
+IPV4LENGTH: int
+IPV6LENGTH: int
 
-class AddressValueError(ValueError): ...
-class NetmaskValueError(ValueError): ...
+_A = TypeVar("_A", IPv4Address, IPv6Address)
+_N = TypeVar("_N", IPv4Network, IPv6Network)
+_T = TypeVar("_T")
 
-class _TotalOrderingMixin:
-    def __eq__(self, other): ...
-    def __ne__(self, other): ...
-    def __lt__(self, other): ...
-    def __le__(self, other): ...
-    def __gt__(self, other): ...
-    def __ge__(self, other): ...
+def ip_address(address: object) -> Union[IPv4Address, IPv6Address]: ...
+def ip_network(address: object, strict: bool = ...) -> Union[IPv4Network, IPv6Network]: ...
+def ip_interface(address: object) -> Union[IPv4Interface, IPv6Interface]: ...
 
-class _IPAddressBase(_TotalOrderingMixin):
+class _IPAddressBase:
+    def __init__(self, address: object) -> None: ...
+    def __eq__(self, other: Any) -> bool: ...
```

Comment:
Why aren't these on `_BaseAddress`?

Reply:
I am not sure I completely understand the question, but I'll attempt an answer anyway. `_IPAddressBase` is the base class of `IPv[46](Address|Interface|Network)`. The attributes shared by those six classes are defined on this class. Although, `__init__` could probably be moved to `_BaseAddress`. I know it doesn't perfectly match the implementation, but since the implementation is private, the result is the same with less code here in the stub.